### ISSUE TITLE
RuntimeException: Cannot add a child that doesn't have a YogaNode to a parent without a measure function! (Trying t...

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-Android#14850]
**Sentry Url**: https://sentry.io/share/issue/f9dc87056c7d4268873cb1b92072fb1b/
**User Count**: 117
**Count**: 172
**First Release**: org.wordpress.android@17.5-rc-1+1056
**First Seen**: 2021-06-01T13:27:28.998000Z
**Last Seen**: 2021-06-16T10:29:04Z
**24 Hours**: 130
**30 Days**: 172

```
RuntimeException: Cannot add a child that doesn't have a YogaNode to a parent without a measure function! (Trying to add a '[RCTVirtualText 13353]' to a '[RCTView 13363]')

...
(4 additional frame(s) were not displayed)
```

**Comment by fluiddot** [comment]
After investigating this issue further, we couldn't manage to reproduce it, here are some insights we gathered along the process:

## Common patterns
The following patterns were found after analyzing Sentry events:

- Many events were triggered by new users.
- Issue happens most often (or always) with **pages** (not posts).
- A volley network error when fetching xposts, is strange because the app shouldn't try to fetch xpost suggestions for -non-P2Blog sites.
- A network error when fetching post formats: `e/WordPress-POSTS: An error occurred while updating the post formats with type: INVALID_RESPONSE`.
- Issue happens most often (or always) when the page is **updated**.

## Potential crash sources
After searching similar issues using the error message, we found the following potential crash sources, all of them related to rendering malformed JSX.

- Comments `//` inside a render function's return statement
- A condition in a render function that resolves to `0` (or `undefined` or an empty string)
- Space between components e.g. `<View> <Text>`
- Raw string value inside a `<Button>` (strings need to be wrapped in a `<Text>`)

## Tests
Since this is a regression, we tested the [Gutenberg Mobile changes introduced in `17.5`](https://github.com/wordpress-mobile/gutenberg-mobile/releases/tag/v1.54.0), however, we couldn't directly relate the crash to any change.

Apart from that we also tried the following scenarios for reproducing the crash but none of them produced it:
- **Use the post/page content from Sentry events:** We've seen events with just a couple of blocks so probably is not related to the content.
- **Create a page using different templates:** As almost all the crashes are related to pages, we explored if it could be related to a specific template but it doesn't.
- **Add different blocks:** Another option was that the crash was related to a specific block, after adding and modifying different blocks we didn't manage to produce the crash.
- **Force the crash by modifying the code:** After running out of options, we decided to force the crash manually to validate the potential crash sources. Surprisingly, the error we got was a different one (`Invariant Violation: Text strings must be rendered within a <Text> component.`) so looks like it might be caused by a different source that we haven't figured out yet.

## User reports

@guarani looked for user reports from the users that most have experienced the crash but none was found related to this.

## Next steps

As per the Sentry events, the issue is happening most often (or always) when the page is **updated** so it would be interesting to continue exploring this hypothesis.

<hr>

Thank you very much @guarani @geriux @oguzkocer @hypest for all the help on this issue 🙇 !

**Comment by antonis** [comment]
I revisited the issue today. After browsing some events I noticed that [in some cases](https://sentry.io/share/issue/f9dc87056c7d4268873cb1b92072fb1b/) the issue occurred after a `LOW_MEMORY` warning. I tried to explore this by stress testing the editor (low memory, don't keep activities flag etc) but was unable to get to the reported crash.

**Comment by fluiddot** [comment]
I found a [related issue](https://github.com/facebook/react-native/issues/13243) that [was fixed](https://github.com/facebook/react-native/issues/13243#issuecomment-572266841) in the version [`0.63.0` of React native](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#fixed-7), I tried to reproduce some of the cases described there but still, I couldn't manage to reproduce this issue.

Nevertheless, looks like it's no longer happening in the beta versions (`17.7-rc-1` and `17.7-rc-2`) of WordPress-Android `17.7` which casually includes the React native upgrade to `0.64`, so some of the fixes from the new versions of React native might have addressed it.

So far we only have 3k sessions on the `17.7` beta versions so we can't really guarantee that it's 100% fixed but it's looking good 🤞.

**Comment by SiobhyB** [comment]
To follow up on Carlos' investigation, Sentry has no reports of this crash tagged as 17.7 or later. As we're two weeks into the release now, I think it's safe to say this is resolved with the upgrade to `0.64`. I'll go ahead to close this but feel welcome to reopen if there's a need to!
